### ISSUE TITLE
Style/Next truncates previous line when indenting is off

### ISSUE THREAD

**Issue by andrewngo** [bbatsov/rubocop#4392]
Running auto-correct on our files. Narrowed down this issue to the `Style/Next` cop. If the `if/end` block is not properly indented, when it changes to `next`, the line above the previous 'end' is truncated by however much the indenting was off.

--------

## Expected behavior
Expect an auto-correction and indenting is correct

```ruby
def test_title(opts = nil)
  [1, 2, 3, 4].each do |num|
    next unless !opts.nil?
    puts num
    if num != 2
      puts 'hello'
      puts 'world'
    end
  end
end
```

## Actual behavior
Since the indenting of `end` was not aligned correctly with its `if`, when it changes to next, it truncates the last letter of the previous line. In this case, causing a syntax error

```ruby
def test_title(opts = nil)
  [1, 2, 3, 4].each do |num|
    next unless !opts.nil?
    puts num
    if num != 2
      puts 'hello'
      puts 'world'
    en
  end
end
```

## Steps to reproduce the problem
```ruby
def test_title(opts = nil)
  [1, 2, 3, 4].each do |num|
    if !opts.nil?
      puts num
      if num != 2
        puts 'hello'
        puts 'world'
      end
   end
  end
end
```

`$rubocop --only Next -a test_file.rb`

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.2.4 x64-mingw32)
```
